slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/validator/GenesisValidatorRootValidator.java

```diff
@@ -36,7 +37,8 @@ public class GenesisValidatorRootValidator {
   private final MetadataDao metadataDao;
   private final FailsafeExecutor<Object> failsafeExecutor;
 
-  private Bytes32 cachedGenesisValidatorRoot;
+  private final AtomicReference<Bytes32> cachedGVRRef = new AtomicReference<>();
```

Comment:
Looks like we are using our cache, think it would be better to use the Guava cache rather than writing our thread safe implementation

Reply:
Hmm, correct. Guava Cache would achieve the same result.

Reply:
or ConcurrentHashMap?